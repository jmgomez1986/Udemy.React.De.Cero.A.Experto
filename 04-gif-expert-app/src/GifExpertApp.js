import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GiftGrid } from './components/GiftGrid';

const GifExpertApp = () => {
	// const categorias = [ 'One Punch', 'Samurai X', 'Dragon Ball'];
	const [categorias, setCategorias] = useState(['One Punch']);

	// const handleAdd = () => {
	// 	// setCategorias([...categorias, 'Nueva Categoria']);
	// 	setCategorias(categories => [...categories, 'Nueva Categoria']);
	// };

	return (
		<>
			<h2>GifExpertApp!!!</h2>
			<AddCategory setCategorias={setCategorias} />

			<hr />

			{/* <button onClick={handleAdd}>Add Category</button> */}

			<ul>
				{
				categorias.map((category) => (
					<GiftGrid
						key={category}
						category={category}
					/>
				))
				}
			</ul>
		</>
	);
};

export default GifExpertApp;
