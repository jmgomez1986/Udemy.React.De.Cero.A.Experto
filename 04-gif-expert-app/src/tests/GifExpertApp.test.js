import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en el componente <GifExpertApp />', () => {
	test('debe de mostrar el componente <GifExpertApp /> correctamente ', () => {
		const wraper = shallow(<GifExpertApp />);

		expect(wraper).toMatchSnapshot();
	});

	test('debe de mostrar una lista de categorias ', () => {
		const categories = ['One Punch', 'Dragon Ball'];
		const wraper = shallow(<GifExpertApp defaultCategories={categories}/>);

		expect(wraper).toMatchSnapshot();
		expect(wraper.find('GifGrid').length).toBe(categories.length );
	});	
	
});
