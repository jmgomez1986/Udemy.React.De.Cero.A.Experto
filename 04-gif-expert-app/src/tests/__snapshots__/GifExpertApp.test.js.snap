// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp /> debe de mostrar el componente <GifExpertApp /> correctamente  1`] = `
<Fragment>
  <h2>
    GifExpertApp!!!
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;

exports[`Pruebas en el componente <GifExpertApp /> debe de mostrar una lista de categorias  1`] = `
<Fragment>
  <h2>
    GifExpertApp!!!
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ul>
</Fragment>
`;
